### PR TITLE
Improving table column resize performance

### LINKED ISSUE #13097: Improve table column resize performance
## 📝 Provide a description of the improvement

When I was researching performance problems with loading data, I stumbled upon table column resize. The plugin is written in the way that it is very non-performant. The initialization takes a lot of time. For some samples, it crashes with `Maximum call stack size exceeded` (and it surely shouldn't the sample wasn't that big). For other, it was heavily slowing down the data load (init and paste).

Right now, whenever view is rendered (after every character type, for example) we go through the whole view document tree and check whether some cells are missing a resizer. It's even worse, because whenever we find such cell, we add the resizer ... triggering another render, so we go again... This is why this caused a crash for pasting a huge table but it would also hinder editor UX when there is a lot of content (even with no tables!).

Since we don't have a nice way to check what changed in the view (or I don't remember such way), for efficiency purposes, I propose to add those UI elements on downcast instead. I already wrote a proposal for a solution but I didn't test it thoroughly.

Steps to do are:

1.  Check manually whether it works correctly.
2.  Write/rework unit tests, if needed.

```javascript
_registerResizerInserter() {
    this.editor.conversion.for( 'editingDowncast' ).add( dispatcher => {
        dispatcher.on( 'insert:tableCell', ( evt, data, conversionApi ) => {
            const mapper = conversionApi.mapper;
            const modelElement = data.item;
            const viewElement = mapper.toViewElement( modelElement );

            if ( !viewElement ) {
                return;
            }

            const viewWriter = conversionApi.writer;
            const viewTableColumnResizerElement = viewWriter.createUIElement( 'div', { class: 'ck-table-column-resizer' } );

            viewWriter.insert(
                viewWriter.createPositionAt( viewElement, 'end' ),
                viewTableColumnResizerElement
            );
        }, { priority: 'lowest' } );
    } );
}
```

### PR DESCRIPTION
**Pull request by DawidKossowski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (table): Adding table resizer elements on editing downcast to improve the performance. Closes #13097.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Comment by scofalik** [comment]
LGTM. AFAIU, you did not need to change any unit test and they are still 100% CC and the change is in a private method. Adding @mlewand for extra review.

**Comment by mlewand** [comment]
@Dumluregn can you look at it?

File: packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js [hunk 1]
```diff
@@ -715,18 +715,15 @@ export default class TableColumnResizeEditing extends Plugin {
...
-			}
-		}, { priority: 'lowest' } );
+		this.editor.conversion.for( 'editingDowncast' ).add( dispatcher => {
+			dispatcher.on( 'insert:tableCell', ( evt, data, conversionApi ) => {
+				const mapper = conversionApi.mapper;
+				const modelElement = data.item;
+				const viewElement = mapper.toViewElement( modelElement );
+				const viewWriter = conversionApi.writer;
+
+				ensureColumnResizerElement( viewWriter, viewElement );
```
[the remark this answers is not in the document]
Comment (scofalik) [reply packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js:725]: I'd even shorten it to this:

```javascript
viewWriter.insert(
	viewWriter.createPositionAt( viewCell, 'end' ),
	viewWriter.createUIElement( 'div', { class: 'ck-table-column-resizer' } );
);
```

File: packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js [hunk 2]
```diff
@@ -715,18 +715,15 @@ export default class TableColumnResizeEditing extends Plugin {
...
-				}
-
-				view.change( viewWriter => {
-					ensureColumnResizerElement( viewWriter, item.item );
-				} );
-			}
-		}, { priority: 'lowest' } );
+		this.editor.conversion.for( 'editingDowncast' ).add( dispatcher => {
+			dispatcher.on( 'insert:tableCell', ( evt, data, conversionApi ) => {
+				const mapper = conversionApi.mapper;
```
Comment (Dumluregn) [inline packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js:720]: Just a nitpick - I'm not a great fan of creating constants that only skip a level of scope and are only used once, with unchanged name even. I'd rather have `const viewElement = conversionApi.mapper.toViewElement( modelElement );` and one `const` less.

`modelElement` is similar but it at least adds some semantics, so I don't have a problem with that.

**Review by Dumluregn — APPROVED** [review]
I've tested various scenarios and didn't find any issue. I tried had to recall why `render` event was used here and I think it was necessary before a huge feature refactor we did half a year ago, and we didn't figure we can improve it then. Anyway, nice idea @scofalik.

I've added one small comment but it's R+ from me after applying Szymon's note.

**Comment by DawidKossowski** [comment]
Done

**Pull request merged**